smack-tcp/src/main/java/org/jivesoftware/smack/tcp/XMPPTCPConnection.java

```diff
@@ -1172,6 +1169,9 @@ private void parsePackets() {
                             if (!stanzasToResend.isEmpty()) {
                                 requestSmAcknowledgementInternal();
                             }
+                            // Mark SM as enabled and resumption as successful.
```

Comment:
I think this is too late. It has probably go to right before the `sendStanzaInternal()` for-loop a few lines above, because `sendStanzaInternal()` calls `isSmEnabled()` which now would return `false. Thus, as far as I can tell know, the right place to mark the SM sync points as successful is *after* we drained the `unacknowledgedStanzas` but *before* we send them using `sendStanzaInternal()`.

Reply:
You are right, I didn't notice this. But in this case new stanzas may be sent before unacknowledged stanzas. May be we should mark `smEnabledSyncPoint` as successful _before_ resending stanzas so `isSmEnabled()` will return `true` and mark `smResumedSyncPoint` _after_ so stanzas will always be sent in the correct order.